Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndo.cpp

```diff
@@ -88,8 +88,23 @@ namespace AzToolsFramework
             //generate undo/redo patches
             m_instanceToTemplateInterface->GeneratePatch(m_redoPatch, initialState, endState);
             m_instanceToTemplateInterface->AppendEntityAliasToPatchPaths(m_redoPatch, entityId);
+            
             m_instanceToTemplateInterface->GeneratePatch(m_undoPatch, endState, initialState);
             m_instanceToTemplateInterface->AppendEntityAliasToPatchPaths(m_undoPatch, entityId);
+
+            AZStd::string entityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(entityId);
+            if (!entityAliasPath.empty())
+            {
+                PrefabDomReference cachedDom = instance.GetCachedInstanceDom();
+
+                // Create a copy of the dom of the end state so that it shares the lifecycle of the cached Dom.
+                PrefabDom endStateCopy;
+                endStateCopy.CopyFrom(endState, cachedDom->get().GetAllocator());
+                Prefab::PrefabDomPath entityPathInDom(entityAliasPath.c_str());
+
+                // Update the cached instance dom corresponding to the entity so that the same modified entity isn't reloaded again.
+                entityPathInDom.Set(cachedDom->get(), AZStd::move(endStateCopy));
+            }
```

Comment:
Suggestion: Instead of adding a GetCachedInstanceDom() for editing the cached dom, could we add a method to the PrefamDom which would do the updating for use? That may be more intuitive since we don't know what kind of changes one would make via GetCachedInstanceDom().

Reply:
Question: Should this be done outside of the undo/redo system? An entity property change in theory could be undoable in which case this code wouldn't get hit.

Reply:
Question: Should this be done outside of the undo/redo system? An entity property change in theory could be undoable in which case this code wouldn't get hit.

Reply:
> Should this be done outside of the undo/redo system? An entity property change in theory could be undoable in which case this code wouldn't get hit.

I see. But why can't we just update the cached dom in `Redo/Undo()` (before PatchTemplate) instead of in `Capture()`? That way we can always keep the consistency.